modules/javafx.graphics/src/main/java/javafx/animation/Animation.java

```diff
@@ -976,18 +992,17 @@ public void playFromStart() {
      *  animation.play();<br>
      * </code>
      * <p>
-     * Note: <ul>
-     * <li>{@code play()} is an asynchronous call, the {@code Animation} may not
-     * start immediately. </ul>
-     * <p>
-     * This method must be called on the JavaFX Application thread.
+     * Note: if this method is not called on the JavaFX Application Thread, it is delegated to it automatically.
+     * In this case, the call is asynchronous and may not happen immediately.
      *
-     * @throws IllegalStateException if this method is called on a thread
-     *                other than the JavaFX Application Thread, or if embedded in another animation,
+     * @throws IllegalStateException if embedded in another animation,
      *                such as {@link SequentialTransition} or {@link ParallelTransition}
      */
     public void play() {
-        Toolkit.getToolkit().checkFxUserThread();
+        Utils.runOnFxThread(this::playOnFxThread);
+    }
+
+    private void playOnFxThread() {
         if (parent != null) {
             throw new IllegalStateException("Cannot start when embedded in another animation");
         }
```

Comment:
Hm, this is going to look really ugly if the caller wasn't on the FX thread.  The exception won't be thrown, it seems to work, but once run the FX thread, it will just log this instead of breaking the callers flow.

Reply:
I suppose you can use this as a "trick" to test if the code to shunt it to the FX thread works (if you don't get the exception immediately, it worked...)

Reply:
Perhaps a better solution is to move this (and similar) checks back to the `play` method, add the checks to the `playFrom*` methods, and do the same for `pause` and `stop`.